openraft/tests/fixtures/mod.rs

```diff
@@ -913,12 +941,20 @@ where
     S: Default + Clone,
 {
     type Network = RaftRouterNetwork<C, S>;
+    type ConnectError = NetworkError;
 
-    async fn connect(&mut self, target: C::NodeId, _node: Option<&Node>) -> Self::Network {
-        RaftRouterNetwork {
+    async fn connect(&mut self, target: C::NodeId, _node: Option<&Node>) -> Result<Self::Network, NetworkError> {
+        let e = NetworkError::new(&AnyError::error(format!("isolated: {}", target)));
+        {
+            let guard = self.isolated_nodes.lock().unwrap();
+            if guard.contains(&target) {
+                return Err(e);
+            }
```

Comment:
change `isolated_nodes` for testing. I am not willing to implement the functionality of `isolated_nodes` for `unreachable_nodes` again.